utils/src/blockuntilzero.rs

```diff
@@ -0,0 +1,110 @@
+use atomic_traits::{Atomic, NumOps};
+use num_traits::{One, Zero};
+use std::{
+    sync::{atomic::Ordering, Arc},
+    time::Duration,
+};
+
+use crate::counttracker::CountTracker;
+
+pub struct BlockUntilZero<T>
```

Comment:
Why are `BlockUntilZero` and `CountTracker` generic over the atomic data type? They already have a very specific task so generalizing them over the underlying data type doesn't make sense to me. Furthermore `usize`/`isize` is a natural choice for storing a size/count of something so I think it would make much more sense for these objects to use `AtomicUsize`/`AtomicIsize` as the type. Now the generic argument requires us to add two new crates while the value proposition of `BlockUntilZero<AtomicU8>` or `BlockUntilZero<AtomicI64>` is unclear to me.

Reply:
Honestly the thing that pissed me off and made me go with generic initially was the lack of `int_fast32_t` alias in rust. And given that I didn't want to think about it and that we want to isolate this to a library for various purposes (which Lukas emphasized even more that I created the BlockUntilZero), I went with a generic to avoid having the discussion "why did you choose 32-bit, and not 16? We'll never have more than 16k events in parallel"... I just don't care about the size thing. I'm sorry; I don't have a good answer for you.